src/components/Tasks/TaskList.tsx

```diff
@@ -126,10 +127,12 @@ export default function TaskList(props: PropsType) {
         }).search(searchTerm);
         return result.map((x) => x.item);
       } else {
-        return props.entries.filter((x) => showCompleted || !x.finished);
+        const notCompleted = (x: TaskType) => showCompleted || !x.finished;
+        const notHidden = (x: TaskType) => showHidden || !x.hidden;
+        return props.entries.filter((x) => notCompleted(x) && notHidden(x));
```

Comment:
Regarding the `notCompleted` and `notHidden`: as I told you before, we prefer named functions over anonymous ones.
Though in this case in particular, I think just put both inside the `.filter` directly rather than having separate functions. I'm sure/hope V8 is smart enough to inline these functions anyway, but since separating them doesn't add any clarity anyway, may as well just inline them yourself.

Reply:
Like this?
```js
return props.entries.filter((x) => (showCompleted || !x.finished) && (showHidden || !x.hidden));
```

Reply:
Yup, exactly.